app/controllers/shifts_controller.rb

```diff
@@ -25,6 +25,10 @@ def new
   end
 
   def edit
+    unless @shift.shift_open?
```

Comment:
Add a similar guard to the actual `shift#update` action in this controller as well 🙏🏾 
What you added here just covers the Edit Form, but not the update action.

Reply:
Thanks for the catch @msarit, have made this a before_action that covers edit, update, and destroy actions